tls/s2n_config.c

```diff
@@ -1071,3 +1072,28 @@ int s2n_config_set_recv_multi_record(struct s2n_config *config, bool enabled)
 
     return S2N_SUCCESS;
 }
+
+/* Indicates if the connection should attempt to enable kTLS. */
+int s2n_config_set_ktls_mode(struct s2n_config *config, s2n_ktls_mode ktls_mode)
+{
+    POSIX_ENSURE_REF(config);
+
+    switch (ktls_mode) {
+        case S2N_KTLS_MODE_DUPLEX:
+            config->ktls_recv_requested = true;
+            config->ktls_send_requested = true;
+            break;
+        case S2N_KTLS_MODE_SEND:
+            config->ktls_send_requested = true;
+            break;
+        case S2N_KTLS_MODE_RECV:
+            config->ktls_recv_requested = true;
+            break;
```

Comment:
https://github.com/aws/s2n-tls/pull/3797/files#r1094771980 I agree that this seems like kind of odd behavior. Why do we need S2N_KTLS_MODE_DUPLEX if we can just call the method twice to enable both send and receive? Maybe these should be "S2N_KTLS_MODE_SEND_ONLY" and "S2N_KTLS_MODE_RECV_ONLY" and set the other direction to false?

Reply:
updated https://github.com/aws/s2n-tls/pull/3797/commits/f0170b4f6adb44f24513c0c121a023f548138ecc